lib/geminabox/gem_list_merge.rb

```diff
@@ -1,23 +1,38 @@
+require "set"
+
 module Geminabox
   class GemListMerge
     attr_accessor :list
 
     IGNORE_DEPENDENCIES = 0..-2
 
-    def self.from(*lists)
-      lists.map{|list| new(list)}.inject(:merge)
+    def self.merge(strategy, lists)
+      lists.map{|list| new(list)}.inject{|a,b| a.merge(b, strategy) }
```

Comment:
This becomes a lot easier if you add `strategy` to the `#initialize` method rather than the `#merge` method.

I'd also pass the strategy in as a named parameter so that it clearly separated from the lists. It would also provide a clear place to define the expected default behaviour.
```ruby
def self.merge(*lists, strategy: :ours)
  lists.map{|list| new(strategy, list)}.inject(:merge)
end
```
Also `attr_accessor` will need to be updated to include `:strategy`

Reply:
Sure; can change style to suit.